src/openforms/formio/components/vanilla.py

```diff
@@ -311,6 +311,12 @@ def rewrite_for_request(component: FileComponent, request: Request):
                 }
             )
 
+    def build_serializer_field(self, component: FileComponent) -> serializers.Field:
+        return serializers.ListField(
+            max_length=component.get("maxNumberOfFiles"),
+            child=serializers.DictField(),
+        )
```

Comment:
this definitely has issues with the `component["validate"]["required"]` key and extensive unit tests are needed for all those cases!

We need to see variants with `multiple: true | false` and other possible combinations too.

Happy flow tests are also essential, not just tests that validate that it's invalid.